sql/moz-fx-data-marketing-prod/ga_derived/downloads_with_attribution_v1/query.sql

```diff
@@ -0,0 +1,315 @@
+    ANY_VALUE(device.operatingSystem) AS os,
+    ANY_VALUE(device.browser) AS browser,
+    ANY_VALUE(device.browserVersion) AS browser_version,
+    ANY_VALUE(device.language) AS `language`,
+    ANY_VALUE(totals.timeOnSite) AS time_on_site
+  FROM
+    `moz-fx-data-marketing-prod.65789850.ga_sessions_*` AS ga
+  WHERE
+    _TABLE_SUFFIX = FORMAT_DATE('%Y%m%d', @submission_date)
+  GROUP BY
+    client_id,
+    visit_id
+),
+ga_sessions_with_hits AS (
+  SELECT
+    *
+  FROM
+    ga_sessions ga
+  JOIN
+    pageviews p
+  USING
+    (client_id, visit_id)
+  JOIN
+    dl_events
+  USING
+    (client_id, visit_id)
+),
+stub AS (
+  SELECT
+    s.jsonPayload.fields.visit_id AS stub_visit_id,
+    s.jsonPayload.fields.dltoken AS dltoken,
+    (COUNT(*) - 1) AS count_dltoken_duplicates,
+    DATE(s.timestamp) AS download_date
+  FROM
+    `moz-fx-stubattribut-prod-32a5.stubattribution_prod.stdout` s
+  WHERE
+    DATE(s.timestamp) = @submission_date
+    AND s.jsonPayload.fields.log_type = 'download_started'
+  GROUP BY
+    stub_visit_id,
+    dltoken,
+    DATE(s.timestamp)
+),
+-- This will drop all the ga_sessions w/o a DLtoken but keep DLtoken without a GA session.
+-- This will also result in multiple rows is the ga.client_id is not unique for the day
+-- since this visit_id is missing from the stub.
+downloads_and_ga_session AS (
+  SELECT
+    gs.client_id AS client_id,
+    ANY_VALUE(country) AS country,
+    ANY_VALUE(ad_content) AS ad_content,
+    ANY_VALUE(campaign) AS campaign,
+    ANY_VALUE(medium) AS medium,
+    ANY_VALUE(source) AS source,
+    ANY_VALUE(device_category) AS device_category,
+    ANY_VALUE(os) AS os,
+    ANY_VALUE(browser) AS browser,
+    ANY_VALUE(browser_version) AS browser_version,
+    ANY_VALUE(`language`) AS `language`,
+    s.stub_visit_id AS stub_visit_id,
+    s.dltoken AS dltoken,
+    ANY_VALUE(landing_page) AS landing_page,
+    ANY_VALUE(pageviews) AS pageviews,
+    ANY_VALUE(unique_pageviews) AS unique_pageviews,
+    LOGICAL_OR(
+      has_ga_download_event
+    ) AS has_ga_download_event,  -- this will be ignored if nrows >1
+    ANY_VALUE(count_dltoken_duplicates) AS count_dltoken_duplicates,
+    COUNT(*) AS nrows,
+    ANY_VALUE(s.download_date) AS download_date,
+    ANY_VALUE(time_on_site) AS time_on_site
+  FROM
+    ga_sessions_with_hits gs
+  RIGHT JOIN
+    stub s
+  ON
+    gs.client_id = s.stub_visit_id
+  GROUP BY
+    gs.client_id,
+    dltoken,
+    stub_visit_id
+)
+SELECT
+  dltoken,
+  download_date,
+  CASE
+    WHEN nrows > 1
+      THEN NULL
+    WHEN stub_visit_id = ''
+      THEN NULL
+    WHEN stub_visit_id = '(not set)'
+      THEN NULL
+    WHEN stub_visit_id = 'something'
+      THEN NULL
+    WHEN client_id IS NULL
+      THEN NULL
+    --only set to null when we do not have a session.  The value can be null when we do have a session so set those to 0.
+    ELSE IFNULL(time_on_site, 0)
+  END
+  time_on_site,
+  CASE
+    WHEN nrows > 1
```

Comment:
If I'm understanding correctly, these fields will be `NULL` for any users with more than one session on that day?

Reply:
Got it. Help me understand what we're going to be using the v1 table for then? Shouldn't we wait until session_id matching is enabled?

Reply:
The priority was to support the session_id matching however with the delayed implementation due to the inconsistent GA session value `_v1` will be released instead.  Since the source data is only stored for 90 days we can use `_v1` to have some representation of a full year of data.